lib/identity_cache/memoized_cache_proxy.rb

```diff
@@ -30,34 +30,47 @@ def with_memoization(&block)
     end
 
     def write(key, value)
-      memoized_key_values[key] = value if memoizing?
-      @cache_fetcher.write(key, value)
+      memoizing = memoizing?
+      ActiveSupport::Notifications.instrument('identity_cache.backend.write', memoizing: memoizing) do
+        memoized_key_values[key] = value if memoizing
+        @cache_fetcher.write(key, value)
+      end
     end
 
     def delete(key)
-      memoized_key_values.delete(key) if memoizing?
-      result = @cache_fetcher.delete(key)
-      IdentityCache.logger.debug { "[IdentityCache] delete #{ result ? 'recorded'  : 'failed'  } for #{key}" }
-      result
+      memoizing = memoizing?
+      ActiveSupport::Notifications.instrument('identity_cache.backend.delete', memoizing: memoizing) do
+        memoized_key_values.delete(key) if memoizing
+        result = @cache_fetcher.delete(key)
+        IdentityCache.logger.debug {"[IdentityCache] delete #{ result ? 'recorded' : 'failed'  } for #{key}"}
+        result
+      end
     end
 
     def fetch(key)
       used_cache_backend = true
       missed = false
-      value = if memoizing?
-        used_cache_backend = false
-        memoized_key_values.fetch(key) do
-          used_cache_backend = true
-          memoized_key_values[key] = @cache_fetcher.fetch(key) do
+      memoizing = memoizing?
+      value = ActiveSupport::Notifications.instrument('identity_cache.backend.fetch', { memoizing: memoizing }) do |payload|
+        value = if memoizing
+          used_cache_backend = false
+          memoized_key_values.fetch(key) do
+            used_cache_backend = true
+            memoized_key_values[key] = @cache_fetcher.fetch(key) do
+              missed = true
+              yield
+            end
+          end
+        else
+          @cache_fetcher.fetch(key) do
             missed = true
             yield
           end
         end
-      else
-        @cache_fetcher.fetch(key) do
-          missed = true
-          yield
-        end
+        payload[:memoized] = used_cache_backend ? 0 : 1
+        payload[:miss] = missed ? 1 : 0
+        payload[:populated] = (missed && value) ? 1 : 0
```

Comment:
This looks like a C-like pattern for booleans.  Why not just use ruby boolean values?